cpp/oneapi/dal/detail/mpi/communicator.hpp

```diff
@@ -266,6 +267,35 @@ class mpi_communicator_impl : public via_host_interface_selector<MemoryAccessKin
         }
     }
 
+    spmd::request_iface* send_receive_replace(byte_t* buf,
+                                              std::int64_t count,
+                                              const data_type& dtype,
+                                              std::int64_t destination_rank,
+                                              std::int64_t source_rank) override {
+        ONEDAL_ASSERT(destination_rank >= 0);
+        ONEDAL_ASSERT(source_rank >= 0);
+
+        if (count == 0) {
+            return nullptr;
+        }
+
+        ONEDAL_ASSERT(buf);
+        ONEDAL_ASSERT(count > 0);
+
+        MPI_Status status;
+        constexpr int zero_tag = 0;
+        mpi_call(MPI_Sendrecv_replace(buf,
+                                      integral_cast<int>(count),
```

Comment:
Isn't it `MPI_INT`?

Reply:
int MPI_Sendrecv_replace(void *buf, int count, MPI_Datatype datatype,
                         int dest, int sendtag, int source, int recvtag,
                         MPI_Comm comm, MPI_Status * status)